_posts/2023-12-13-release-0_4_0.md

```diff
@@ -0,0 +1,26 @@
+---
+layout: post
+title:  "kroxylicious release 0.4.0"
+date:   2023-12-14 00:00:00 +1300
+author: "Keith Wall"
+author_url: "https://www.github.com/k-wall"
+categories:  releases kroxylcious topic-encryption
+---
+
+The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.4.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.4.0).
+
+The big news is this release includes our `EnvelopeEncryption` filter which provides an **Encryption-at-Rest** solution for Apache Kafka(tm) which is _transparent_ to both clients and brokers. We have more work planned to iterate on the implementation to reach production quality. Check out our [0.5.0 milestone](https://github.com/kroxylicious/kroxylicious/milestones/0.5.0) for details.
+
+The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and encrypts the Kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
+
+The Kafka Cluster *never* sees the plain text of your records. Filter configuration is used to specify which topic(s) should be encrypted by which key.
+
+The encryption keys themselves are stored safely in a Key Management System.  0.4.0 ships with a [HashiCorp Vault](https://www.hashicorp.com/products/vault)(tm) integration.  We will provide other selected Key Management System integrations soon, however users can leverage the plug-in architecture of Kroxylicious to supply their own integrations.
+
+The diagram below shows how the system works at a high level.
+
+![image](https://github.com/kroxylicious/kroxylicious.github.io/assets/18440250/09aeb1f4-f420-4d89-b9f7-394473dddc05)
+
+If this use-case interests you, a [short demo](https://asciinema.org/a/627519) that shows the feature in action will help you understand more and there's some [interim instructions[(https://github.com/kroxylicious/kroxylicious/blob/f5f62a48d1db67309eee1696bb7a70dbb83ba622/kubernetes-examples/envelope-encryption/README.md) to help you try out the feature in development.  There will be more information landing on the website soon, but meanwhile, if you have questions or comments please head over to the [Slack channel](https://kroxylicious.slack.com/).
```

Comment:
My intent is that https://github.com/kroxylicious/kroxylicious/pull/825 is merged then this link it updated to refer to main.    For the next release https://github.com/kroxylicious/kroxylicious/issues/827 should be resolved, so we'd be able to reference the example on tag (hopefully it'll be superseded by quickstart content by then too).